Userland/Libraries/LibJS/Runtime/JSONObject.cpp

```diff
@@ -432,8 +432,6 @@ Value JSONObject::parse_json_value(VM& vm, JsonValue const& value)
         return Value(parse_json_array(vm, value.as_array()));
     if (value.is_null())
         return js_null();
-    if (value.is_i32())
-        return Value(value.as_i32());
```

Comment:
In general, when we have an `i32`, we don't want to round-trip through a `double` for no reason.
This change seems like it introduces a pointless `i32->double->i32` round-trip, no?

Reply:
Oh, I just realized that without the context of `JsonValue` refactoring, this change doesn't make much sense indeed :). I guess I will upstream this together with the full refactoring sometime in the future then.